addons/base/utils.py

```diff
@@ -20,3 +21,28 @@ def serialize_addon_config(config, user):
 def get_addons_by_config_type(config_type, user):
     addons = [addon for addon in settings.ADDONS_AVAILABLE if config_type in addon.configs]
     return [serialize_addon_config(addon_config, user) for addon_config in sorted(addons, key=lambda cfg: cfg.full_name.lower())]
+
+def maybe_show_lost_file_metadata(auth, node, file, error_type):
+    msg = """
+    </div>"""  # None is default
+    if error_type != 'FILE_SUSPENDED' and ((auth.user and node.is_contributor(auth.user)) or (auth.private_key and auth.private_key in node.private_link_keys_active)):
+        last_meta = file.last_known_metadata
+        last_seen = last_meta.get('last_seen', None)
+        hashes = last_meta.get('hashes', None)
+        path = last_meta.get('path', None)
+        size = last_meta.get('size', None)
+        parts = [
+            """</br>This file was """ if last_seen or hashes or path or size else '',
+            """last seen on {} UTC """.format(last_seen.strftime('%c')) if last_seen else '',
+            """and found at path {} """.format(markupsafe.escape(path)) if last_seen and path else '',
+            """last found at path {} """.format(markupsafe.escape(path)) if not last_seen and path else '',
+            """with a file size of {} bytes""".format(size) if size and (last_seen or path) else '',
+            """last seen with a file size of {} bytes""".format(size) if size and not (last_seen or path) else '',
+            """.</br></br>""" if last_seen or hashes or path or size else '',
+            """Hashes of last seen version:</br>{}""".format(
```

Comment:
Nitpick of all nitpicks: If the "hashes" section was wrapped in a `<p>` tag, the hashes would wrap correctly on small screens instead of overflowing.

![screen shot 2018-02-19 at 11 33 47 am](https://user-images.githubusercontent.com/7913604/36388120-dccdd90a-1568-11e8-919f-7bcded572ed4.png)
![screen shot 2018-02-19 at 11 33 02 am](https://user-images.githubusercontent.com/7913604/36388121-dcdcf8e0-1568-11e8-9591-458826202a66.png)